pennylane_lightning_gpu/src/tests/mpi/Test_MPI_StateVectorCudaManaged_Param.cpp

```diff
@@ -0,0 +1,319 @@
+            SECTION("Operation on target") {                                   \
+                StateVectorCudaMPI<TestType> sv(mpi_manager, mpi_buffersize,   \
+                                                nGlobalIndexBits,              \
+                                                nLocalIndexBits);              \
+                sv.CopyHostDataToGpu(local_state, false);                      \
+                sv.GATE_METHOD(WIRE, false, ANGLE);                            \
+                sv.CopyGpuDataToHost(local_state.data(), subSvLength);         \
+                SVDataGPU<TestType> svdat{NUM_QUBITS, init_sv};                \
+                if (mpi_manager.getRank() == 0) {                              \
+                    svdat.cuda_sv.GATE_METHOD(WIRE, false, ANGLE);             \
+                    svdat.cuda_sv.CopyGpuDataToHost(expected_sv.data(),        \
+                                                    svLength);                 \
+                }                                                              \
+                auto expected_local_sv = mpi_manager.scatter(expected_sv, 0);  \
+                CHECK(local_state == Pennylane::approx(expected_local_sv));    \
+            }                                                                  \
+        }                                                                      \
+        SECTION("Apply using dispatcher") {                                    \
+            SECTION("Operation on target") {                                   \
+                StateVectorCudaMPI<TestType> sv(mpi_manager, mpi_buffersize,   \
+                                                nGlobalIndexBits,              \
+                                                nLocalIndexBits);              \
+                sv.CopyHostDataToGpu(local_state, false);                      \
+                sv.applyOperation(GATE_NAME, WIRE, false, ANGLE);              \
+                sv.CopyGpuDataToHost(local_state.data(), subSvLength);         \
+                SVDataGPU<TestType> svdat{NUM_QUBITS, init_sv};                \
+                if (mpi_manager.getRank() == 0) {                              \
+                    svdat.cuda_sv.applyOperation(GATE_NAME, WIRE, false,       \
+                                                 ANGLE);                       \
+                    svdat.cuda_sv.CopyGpuDataToHost(expected_sv.data(),        \
+                                                    svLength);                 \
+                }                                                              \
+                auto expected_local_sv = mpi_manager.scatter(expected_sv, 0);  \
+                CHECK(local_state == Pennylane::approx(expected_local_sv));    \
+            }                                                                  \
+        }                                                                      \
+    }
```

Comment:
Here too. This Macro function could in principle be substituted by a usual function.

Reply:
See above.